tools/lsp/main.rs

```diff
@@ -32,6 +34,51 @@ use std::rc::Rc;
 use std::sync::{atomic, Arc, Mutex};
 use std::task::{Poll, Waker};
 
+struct Previewer {
+    server_notifier: ServerNotifier,
+}
+
+impl PreviewApi for Previewer {
+    fn set_design_mode(&self, enable: bool) {
+        #[cfg(feature = "preview")]
+        preview::set_design_mode(self.server_notifier.clone(), enable);
+    }
+
+    fn design_mode(&self) -> bool {
+        #[cfg(feature = "preview")]
+        preview::design_mode()
+    }
+
+    fn set_contents(&self, path: &std::path::Path, contents: &str) {
+        #[cfg(feature = "preview")]
+        preview::set_contents(path, contents.to_string());
+    }
+
+    fn load_preview(
+        &self,
+        component: common::PreviewComponent,
+        behavior: common::PostLoadBehavior,
```

Comment:
you will need to use `_`  or `allow(unused)`

Reply:
This is not resolved  when the  `#[cfg(feature = "preview")]`  is disabled these parametter will be unused